src/navigation/MainTabNavigator.tsx

```diff
@@ -1,37 +1,114 @@
 import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import { colors, device, normalize } from '../config';
-import { TabNavigatorConfig } from '../types';
+import { LoadingSpinner } from '../components';
+import { colors, normalize } from '../config';
+import { createDynamicTabConfig, tabNavigatorConfig } from '../config/navigation/tabConfig';
+import { useStaticContent } from '../hooks';
+import { TabConfig, TabNavigatorConfig } from '../types';
 
 import { getStackNavigator } from './AppStackNavigator';
 
+const useTabRoutes = () => {
+  const {
+    data: tabRoutesData,
+    error,
+    loading
+  } = useStaticContent<TabNavigatorConfig>({
+    name: 'tabNavigation',
+    type: 'json'
+  });
+
+  const [tabRoutes, setTabRoutes] = useState<TabNavigatorConfig>(tabNavigatorConfig);
+
+  useEffect(() => {
+    const {
+      activeBackgroundColor,
+      activeTintColor,
+      inactiveBackgroundColor,
+      inactiveTintColor,
+      tabConfigs
+    } = tabRoutesData || tabNavigatorConfig;
+
+    setTabRoutes((prev) => {
+      const dynamicTabs = tabConfigs?.map((tab, index) => {
+        if (typeof tab === 'string') {
+          return tabNavigatorConfig.tabConfigs.find(
+            (tabConfig) => tabConfig.stackConfig.initialRouteName === tab
```

Comment:
should this always use `tabNavigatorConfig.tabConfigs` and not `tabConfigs` from `tabRoutesData || tabNavigatorConfig`?

Reply:
Yes. Here we are comparing defaultTabs with the array on main-server. If the string in the main-server array matches the `initialRouteName` in the `tabNavigatorConfig` (default), that tab is automatically selected.

Reply:
why did you only add `// if tab is a string, find the corresponding tabConfig from default config` and not your whole two sentences. please update that.